### PR TITLE
Simplify color interpolation

### PR DESCRIPTION
**Pull request by blair1618**

### CODE REVIEW

File: core/src/util/color.h [hunk 1]
```diff
@@ -9,10 +9,22 @@ struct Color {
...
-        uint32_t abgr = 0x00000000;
-        // channel order appears reversed because of little-endianness
+        uint32_t abgr = 0;
     };
 
+    Color() = default;
+    Color(uint32_t _abgr) : abgr(_abgr) {}
+    Color(uint8_t _r, uint8_t _g, uint8_t _b, uint8_t _a) : r(_r), g(_g), b(_b), a(_a) {}
+
+    static Color mix(const Color& _x, const Color& _y, float _a) {
```
Comment (tallytalwar) [inline core/src/util/color.h]: Instead of mix, can we call it interpolate or something else.

Comment (karimnaaji) [inline core/src/util/color.h]: `mix` is the standard built in name for glsl color interpolation, so it quite makes sense

Comment (tallytalwar) [inline core/src/util/color.h]: then its fine. :D

Comment (blair1618) [inline core/src/util/color.h]: Yep that was the idea

**Comment by blair1618** [comment]
I'll take that to mean there are no issues with this change :)

**Pull request merged**